src/devices_models/devices/common/cost_functions.jl

```diff
@@ -10,6 +10,7 @@ struct AddCostSpec
     shut_down_cost::Union{Nothing, Function}
     fixed_cost::Union{Nothing, Function}
     has_multistart_variables::Bool
+    has_target_cost::Bool
```

Comment:
Why do we need a bool here is there is also the field additional linear terms? we should just add the terms if they exist.